cloud-sql/postgres/Program.cs

```diff
@@ -14,21 +14,55 @@
  * the License.
  */
 
+using System;
+using System.IO;
 using Microsoft.AspNetCore;
 using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Configuration;
+using Google.Cloud.Diagnostics.AspNetCore;
+using CloudSql.Settings;
+
 
 namespace CloudSql
 {
     public class Program
     {
+        public static AppSettings AppSettings { get; private set; }
+
         public static void Main(string[] args)
         {
-            BuildWebHost(args).Run();
+            BuildWebHost(args).Build().Run();
         }
 
-        public static IWebHost BuildWebHost(string[] args) =>
-            WebHost.CreateDefaultBuilder(args)
-                .UseStartup<Startup>()
+        public static IWebHostBuilder BuildWebHost(string[] args)
+        {
+            ReadAppSettings();
+
+            string port = Environment.GetEnvironmentVariable("PORT") ?? "8080";
+            string hostingUrl = String.Concat("http://0.0.0.0:", port);

```

Comment:
nit: don't use String, better use string.
And also, the following suggestion is way more idiomatic:
```suggestion
            string hostingUrl = $"http://0.0.0.0:{port}");

```